tests/build/multi-platform.go

```diff
@@ -411,8 +421,617 @@ var _ = framework.MultiPlatformBuildSuiteDescribe("Multi Platform Controller E2E
 
 		})
 	})
+	Describe("ibmz dynamic allocation", Label("ibmz-dynamic"), func() {
+		var testNamespace, applicationName, componentName, multiPlatformSecretName, multiPlatformTaskName, dynamicInstanceTag, instanceId string
+		var component *appservice.Component
+		var timeout, interval time.Duration
+
+		AfterAll(func() {
+			//Cleanup ibm&ssh secrets and host-config
+			Expect(f.AsKubeAdmin.CommonController.DeleteSecret(ControllerNamespace, IbmSecretName)).To(Succeed())
+			Expect(f.AsKubeAdmin.CommonController.DeleteSecret(ControllerNamespace, SshSecretName)).To(Succeed())
+			Expect(f.AsKubeAdmin.CommonController.DeleteConfigMap(HostConfig, ControllerNamespace, true)).To(Succeed())
+
+			//Forcefully remove instance incase it is not removed by multi-platform-controller
+			err = terminateIbmZInstance(instanceId)
+			if err != nil {
+				GinkgoWriter.Printf("error terminating instance again: %v", err)
+			}
+
+			if !CurrentSpecReport().Failed() {
+				Expect(f.AsKubeAdmin.HasController.DeleteComponent(componentName, testNamespace, false)).To(Succeed())
+				Expect(f.AsKubeAdmin.HasController.DeleteApplication(applicationName, testNamespace, false)).To(Succeed())
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace)).To(Succeed())
+			} else {
+				Expect(f.AsKubeAdmin.CommonController.StoreAllPods(testNamespace)).To(Succeed())
+				Expect(f.AsKubeAdmin.TektonController.StoreAllPipelineRuns(testNamespace)).To(Succeed())
+			}
+		})
+
+		BeforeAll(func() {
+
+			f, err = framework.NewFramework(utils.GetGeneratedNamespace("multi-platform-ibmz"))
```

Comment:
Other than the setup are all these tests pretty much the same? It seems like a lot of duplication here.

Reply:
let me try to reduce the duplicates.